### DISCUSSION TITLE
Git repo commits not syncing to gitbook (Monorepo)

### DISCUSSION THREAD

**Discussion by strazto** [GitbookIO/community#329] [3. Git Sync]
Related: #285 

# Problem: 

In a space configured to sync with GitHub under the monorepo configuration, commits to the relevant branch + project root in GitHub are not being reflected back in GitBook

# Configuration

I connected GitBook to a GitHub repo, using the "Monorepo" feature to specify a subfolder as the content root. 

![config](https://github.com/GitbookIO/community/assets/39424834/58dfe445-2769-4dca-867a-515f03de365c)

I chose GitHub -> GitBook for my initial sync strategy (I forget the specific wording of this initialization option)

My intended usage of GitBook is to host a combination of manually written documents, as well as one or more collections of statically generated API docs that can be generated and committed manually, or perhaps as part of some CI.

# Current Behaviour:

## Change History

The initial github -> GitBook import worked, however subsequent updates / pushes to `./docs/userdocs` have not been reflected in GitBook.

![missed_updates_gitbook](https://github.com/GitbookIO/community/assets/39424834/50b5f4fe-3c80-4a60-a8f9-46db37a8f291)

I've been able to make a CR from gitbook which wrote to the repo with the stale state that gitbook could see - gitbook is able to see this change in the history, however it can't see prior or subsequent changes (besides the initial sync) from GitHub -> GitBook

![full_updates_github](https://github.com/GitbookIO/community/assets/39424834/2e5cb018-b4a9-4e26-9c8e-f78b2846d1c7)

## Document State

The content / state of documents seen in GitBook does not reflect changes made in GitHub.

In the following example, I realized that GitBook doesn't give `yaml` frontmatter special treatment and renders it as if it were markdown, so I removed the frontmatter

GitBook - Still contains now removed `yaml` frontmatter:

![GitBook - Out of date](https://github.com/GitbookIO/community/assets/39424834/5319676b-d5b7-4576-9d32-0d8700da7087)

GitHub - Preview no longer contains removed `yaml` frontmatter:
![image](https://github.com/GitbookIO/community/assets/39424834/be3e8dfd-a26f-406d-b51e-5e9754694173)

# Expected Behaviour

The current state of documents in GitBook matches that seen in the same document on GitHub

The commit history displayed in GitBook is equivalent to the commit history seen in that directory in GitHub.

# Background

This issue is actually documented in #285 ,

> I pushed changes to my remote repo 14 minutes ago, but GitBook hasn't realized yet.

However that discussion was resolved prematurely by the author with a workaround that doesn't constitute a viable solution:

> Nevermind. I delete the space and made a new one. That seemed to solve the problem for me.

**Comment by aikoii** [comment]
Hello @strazto, I'm sorry to hear you're having trouble! It's possible your issue was a temporary one, now that your space is successfully synced. However, if your repo commits are still not syncing to GitBook, we'd have to inspect logs for any errors. Could you please share your space ID? You can email us at [support@gitbook.com](mailto:support@gitbook.com).
